core/Tracker/Visit.php

```diff
@@ -213,6 +214,10 @@ public function handle()
         $this->request->setThirdPartyCookie($this->request->getVisitorIdForThirdPartyCookie());
 
         foreach ($this->requestProcessors as $processor) {
+            if (!$isNewVisit && $processor instanceof ActionsRequestProcessor) {
+                // already processed earlier when handling exisitng visit see {@link self::handleExistingVisit()}
+                continue;
+            }
```

Comment:
could this be put in ActionsRequestProcessor? ie, return if conditions are met?

Reply:
oh i see, this has to be done before updating a visit... seems like it would all be clearer if this was all in ActionsRequestProcessor, though i'm not sure if that's possible

Reply:
I know it's not really clear which is why I'm not sure yet if it's worth merging it. Problem is for new visits we have to execute the actions processor later vs for existing visits we have to execute it before updating a visit...